cfn/service.yml

```diff
@@ -174,24 +174,50 @@ Resources:
       Port: 8000
       Protocol: HTTP
       VpcId: !Ref VpcId
-  RandomLoadBalancerRule:
+  DiscoveryRandomLoadBalancerRule:
     Type: AWS::ElasticLoadBalancingV2::ListenerRule
     Properties:
       Actions:
-      - TargetGroupArn: !Ref 'RandomTargetGroup'
+      - TargetGroupArn: !Ref 'DiscoveryRandomTargetGroup'
         Type: 'forward'
       Conditions:
       - Field: path-pattern
-        Values: ['/random']
-      ListenerArn: !Ref AppLoadBalancerListener
+        Values: ['/api/v1/random/**']
+      ListenerArn: !Ref DiscoveryAppLoadBalancerListener
       Priority: 2
-
+  # When swagger container comes up, it will register in this target group
+  DiscoverySwaggerTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      Name: DISCOVERY-SWAGGER-TG
+      HealthCheckIntervalSeconds: 150
+      HealthCheckPath: /actuator/health
+      HealthCheckPort: 8001
+      HealthCheckProtocol: HTTP
+      HealthCheckTimeoutSeconds: 120
+      HealthyThresholdCount: 2
+      UnhealthyThresholdCount: 2
+      TargetType: ip
+      Port: 8000
+      Protocol: HTTP
+      VpcId: !Ref VpcId
+  DiscoverySwaggerLoadBalancerRule:
+    Type: AWS::ElasticLoadBalancingV2::ListenerRule
+    Properties:
+      Actions:
+      - TargetGroupArn: !Ref 'DiscoverySwaggerTargetGroup'
+        Type: 'forward'
+      Conditions:
+      - Field: path-pattern
+        Values: ['/api/v1/**']
```

Comment:
Can we change path to '/api/v1/swagger/**'?

Reply:
If we change the path to '/api/v1/swagger/**', then we need to add '/api/v1/swagger' as the context path in Swagger service. The challenge here is, when we need to access swagger docs of other services, we define the path of swagger docs for each service in application.yml file of Swagger service. This path gets appended to the context path '/api/v1/swagger'. For example, the path to access swagger docs for welcome service is '/welcome/v2/api-docs'. This path is appended to the context path as '/api/v1/swagger/welcome/v2/api-docs' to get the swagger docs of Welcome service. But on hitting '/api/v1/swagger/welcome/v2/api-docs' url, path-based routing fails in target group as '/api/v1/swagger/welcome' is not the context path of Welcome service. To overcome this issue, we added '/api/v1/**' path in the listener rule and gave last priority, in this case, the priority is 4 and added the context path in Swagger service as '/api/v1'. So that, when we access swagger docs of Welcome service from Swagger service, the path will be built as '/api/v1/welcome/v2/api-docs'.